app/controllers/alchemy/admin/pages_controller.rb

```diff
@@ -25,6 +27,59 @@ def index
         end
       end
 
+      # Returns all pages as a tree from the root given by the id parameter
+      #
+      def tree
+        authorize! :tree, :alchemy_admin_pages
+
+        tree = []
+        path = [{id: @page.parent_id, children: tree}]
+        page_list = @page.self_and_descendants
+
+        page_list.each_with_index do |page, i|
+          if page.parent_id != path.last[:id]
+            if path.map { |o| o[:id] }.include?(page.parent_id) # Lower level
+              path.pop while path.last[:id] != page.parent_id
+            else # One level up
+              path << path.last[:children].last
+            end
+          end
+
+          has_children = page_list[i + 1] && page_list[i + 1].parent_id == page.id
+          level = path.count
+
+          path.last[:children] << {
```

Comment:
Hmm, ok. I thought of `jbuilder` or `rabl`, but this would add another dependency. Interesting background: http://stackoverflow.com/questions/26097563/jbuilder-vs-rails-api-active-model-serializers-for-json-handling-in-rails-4

Instead of concern/helper I would like to have some kind of json presenter class. So we could do something like:

```ruby
Alchemy::PagePresenter.new(page, current_ability).to_json
```